packages/back-end/src/queryRunners/ExperimentResultsQueryRunner.ts

```diff
@@ -279,7 +279,7 @@ export const startExperimentResultQueries = async (
     organization
   );
 
-  const singlePromises = singles.map(async (m) => {
```

Comment:
The ES6 for loop format allows for awaiting within each iteration of the loop without going to the next, so we can drop the concurrency granularity to 1

Reply:
I see. So the plan here is to be in line with Jeremy's comment and not kick off the queries all at once and await the query model to be created but rather do it sequentially.

It's kind of hard to tell, but is there any chance that `startQuery` gets stuck? In that method we call `setTimeout` a few times, as well as `executeQueryWhenReady` and I want to be sure those don't hold up the rest of the queries from being written to the DB asap. Is this a concern or am I just misunderstanding what will actually be `await`ed in `startQuery`?

Reply:
My main concern is that some query gets stuck waiting to start and then that await doesn't let any of the other queries in the model get created and enter the `queued` state because this for loop has paused execution on the queued query. I could be totally off base about this.

Reply:
Yep, that's the plan. For posterity, all of the async logic within `startQuery` is just waiting for reads & writes to Mongo (i.e. all bookkeeping), not anything related to the content of the query itself. So those awaits should be tiny delays & all the queries will probably be kicked off by the time the first one finishes (if the concurrency limit allows)